doc/sphinx/proof-engine/tactics.rst

```diff
@@ -105,6 +105,227 @@ bindings_list`` where ``bindings_list`` may be of two different forms:
+introduced by tactics.  They also let you split an introduced hypothesis into
+multiple hypotheses or subgoals.  Common tactics that accept intro patterns
+include :tacn:`intros`, :tacn:`destruct` and :tacn:`apply`.
+
+.. productionlist::        coq
+   simple_intropattern     : `simple_intropattern_closed`
+                           : % `operconstr` ... % `operconstr`
+   simple_intropattern_closed : `or_and_intropattern`
+                              : `equality_intropattern`
+                              : _
+                              : `naming_intropattern`
+   or_and_intropattern     : [ `intropatterns` | ... | `intropatterns` ]
+                           : ()
+                           : ( `simple_intropattern` , ... , `simple_intropattern` )
+                           : ( `simple_intropattern` & ... & `simple_intropattern` )
+   equality_intropattern   : ->
+                           : <-
+                           : [= `intropatterns` ]
+   intropatterns           : `nonsimple_intropattern` ...
+                           : <empty>
+   ne_intropatterns        : `nonsimple_intropattern` ...
+   nonsimple_intropattern  : `simple_intropattern`
+                           : *
+                           : **
+   naming_intropattern     : `pattern_ident`
+                           : ?
+                           : `ident`
+   pattern_ident           : ? `ident`
+
+Use these elementary patterns to specify a name:
+
+* :n:`@ident` - use the specified name
+* :n:`?` - let Coq choose a name
+* :n:`? @ident` - generate a name that begins with :n:`@ident`
+* :n:`_` - discards the matched part (unless it is required for another
+  hypothesis)
+* if a disjunction pattern omits a name, such as :g:`[|H2]`, Coq will choose a name
+
+The most common splitting patterns are:
+
+* split a hypothesis in the form :g:`A /\ B` into two
+  hypotheses :g:`H1: A` and :g:`H2: B` using the pattern :g:`[H1 H2]`
+* split a hypothesis in the form :g:`A \/ B` into two
+  subgoals using the pattern :g:`[H1|H2]`.  The first subgoal will have the hypothesis
+  :g:`H1: A` and the second subgoal will have the hypothesis :g:`H2: B`.
+
+**:ref:`Example <_intropattern_conj_ex>` doesn't work for linking to the example**
+
+**Sphinx says: "WARNING: undefined label: _intropattern_conj_ex (if the link has no caption
+the label must precede a section header)"**
+
+Splitting patterns can have more than 2 names, for example :g:`[H1 H2 H3]`
+matches :g:`A /\ B /\ C`.  Patterns can be nested:
+:g:`[[Ha|Hb] H]` can be used to split :g:`(A \/ B) /\ C`.
+
+Note that there is no equivalent to intro patterns for goals.  For a goal :g:`A /\ B`,
+use the :tacn:`split` tactic to replace the current goal with subgoals :g:`A` and :g:`B`.
+For a goal :g:`A \/ B`, use :g:`apply or_introl` to replace the current goal with :g:`A`, or
+:g:`apply or_intror` to replace the current goal with :g:`B`.
+
+The other splitting patterns described below are not frequently used.
```

Comment:
On the contrary, as I said before, the once with commas and `&` are very much recommended.

Reply:
Indeed, but the code of the standard library is not a good example of how people would write things today because it is mostly ancient code that has very rarely been updated.